tests/helpers/fake_backend.py

```diff
@@ -51,41 +51,46 @@ def do_POST(self):
             self.end_headers()
             self.wfile.write("\"OK\"".encode("utf-8"))
 
-    return HTTPServer(('127.0.0.1', 0), FakeIngest)
+    return HTTPServer((ip, 0), FakeIngest)
 
 
 # Fake the dimension PUT method to capture dimension property/tag updates.
-def _make_fake_api(dims):
+def _make_fake_api(dims, ip="127.0.0.1"):
     class FakeAPIServer(BaseHTTPRequestHandler):
         def do_PUT(self):
             if '/dimension/' not in self.path:
                 self.send_response(404)
                 self.end_headers()
                 return
 
-            body = self.rfile.read(int(self.headers.getheader('Content-Length')))
+            try:
+                body = self.rfile.read(int(self.headers.getheader('Content-Length')))
```

Comment:
I think I added that when I first started and ran into some incompatibility issues with python2.  I'll revert the change since we should only be using python3.

Reply:
Here's the exception:
```Exception happened during processing of request from ('172.17.0.3', 52592)
Traceback (most recent call last):
  File "/usr/lib/python3.5/socketserver.py", line 313, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 341, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 354, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.5/socketserver.py", line 681, in __init__
    self.handle()
  File "/usr/lib/python3.5/http/server.py", line 422, in handle
    self.handle_one_request()
  File "/usr/lib/python3.5/http/server.py", line 410, in handle_one_request
    method()
  File "/go/src/github.com/signalfx/signalfx-agent/tests/helpers/fake_backend.py", line 66, in do_PUT
    body = self.rfile.read(int(self.headers.getheader('Content-Length')))
AttributeError: 'HTTPMessage' object has no attribute 'getheader'```

It doesn't seem to affect the tests, but I can update it to just capture AttributeError exceptions.

Reply:
Can it just `get`?